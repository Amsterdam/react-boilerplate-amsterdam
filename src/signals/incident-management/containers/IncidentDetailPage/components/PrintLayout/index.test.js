import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { fromJS } from 'immutable';

import ConnectedPrintLayout, { PrintLayout } from './index';

describe('<PrintLayout />', () => {
  let props;

  beforeEach(() => {
    props = {
      id: '100',
      incident: {},
      stadsdeelList: [],
      onPrintView: jest.fn(),
      incidentstatuscontainer: { incidentStatusList: [], statusList: [] }
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <PrintLayout {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the connected component correctly', () => {
    const mockStore = configureMockStore({});
    const store = mockStore(fromJS({ incidentStatusContainer: { incidentStatusList: [], statusList: [] } }));
    const wrapper = shallow(<ConnectedPrintLayout store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

