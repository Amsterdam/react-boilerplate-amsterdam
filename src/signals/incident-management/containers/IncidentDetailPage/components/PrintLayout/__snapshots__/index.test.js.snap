// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrintLayout /> should render correctly 1`] = `
<div
  className="print-layout row container"
>
  <div
    className="col-12"
  >
    <h3>
      Melding 
      100
    </h3>
    <div
      className="no-print"
    >
      <button
        className="no-print"
        onClick={[Function]}
      >
        Print
      </button>
      <button
        className="no-print"
        onClick={[MockFunction]}
      >
        Terug
      </button>
    </div>
  </div>
  <div
    className="col-12 incident-print-page__map"
  >
    <MapDetail
      label=""
    />
  </div>
  <div
    className="col-12"
  >
    <IncidentDetail
      incident={Object {}}
      stadsdeelList={Array []}
    />
  </div>
  <div
    className="col-12"
  />
</div>
`;
